ReservationSystem/LogicLayer/Guide.cs

```diff
@@ -187,4 +187,42 @@ public static void ReassignGuideToTour()
         }
 
     }
+
+    public Tour AddVisitorLastMinute(Visitor visitor)
+    {
+        Tour.LoadTours();  // refresh TodaysTours data.
+        // ussing method to filter specific conditions.
+        // orderd by starttime.
+        var availableGuideTours = Tour.FilterByLambda(tour => tour.AssignedGuide.Name == this.Name
+            && !tour.Completed && !tour.Deleted 
+            && tour.ExpectedVisitors.Count < tour.MaxCapacity)
+            .OrderBy(tour => tour.StartTime).ToList(); 
+        
+        if (availableGuideTours.Count == 0)  // there are no more availble tours for visitor
+        {
+            NoAvailbleTour.Show();
+            return null;  // break out of void to prefent program crash
+        }
+        
+        Tour target = availableGuideTours.First();  // the target will chase down the closest next tour to overwrite // Data with visitor added to PresentVisitor will overwrite the target
+        target.PresentVisitors.Add(visitor);
+        Tour overwite = target;
+
+        // loop trought todays tours and overwite. 
+        for (int index = 0; index < Tour.TodaysTours.Count; index++)
+        {
+            var tour = Tour.TodaysTours[index];
+            if (tour.TourId == target.TourId && tour.AssignedGuide.Name == this.Name)  
+            {
+                Tour.TodaysTours[index] = overwite;  // update the Tour with visitor added to Pressent Visitor
+                Tour.SaveTours();  // overwrite JSON with the visitor added to Tour 
+                Tour.LoadTours();  // refresh TodaysTours data afther changes.
+                return tour;  // break out the method and send tourDetail to display aditional info for guide.
+            }
+        }
```

Comment:
no need for all of this, only add savetours at the end. and everything will be updated automatically. the static lists would be updated the second u put someone in them so no need to manually update them.